### PR TITLE
Update for XAPK support (tweaselORG/appstraction#64)

### PR DESCRIPTION
**Pull request by baltpeter**
This doesn't actually upgrade to the new appstraction version yet as that isn't released yet.

### CODE REVIEW

**Comment by zner0L** [comment]
Is there any reason in particular why this is still in draft state?

**Comment by baltpeter** [comment]
No. GitHub had an issue when I made that changes and they didn't show up until an hour later.

**Review by zner0L — CHANGES_REQUESTED** [review]
It looks mostly great, but as I said in https://github.com/tweaselORG/cyanoacrylate/issues/22#issuecomment-1581589779, we need to add an `await platform.waitForDevice()` in ensure Device because we moves that out of `platform.ensureDevice()` in appstraction v0.3.0.

File: src/index.ts [hunk 1]
```diff
@@ -467,6 +467,7 @@ export async function startAnalysis<
 
                     // eslint-disable-next-line require-atomic-updates
                     emulatorProcess = execa('emulator', cmdArgs);
+                    await platform.waitForDevice();
```
Comment (zner0L) [inline src/index.ts:470]: It turns out that on my Macbook at least, 20 tries are not enough to wait for the emulator to start, I need about 100.

**Review by zner0L — CHANGES_REQUESTED** [review]

File: src/index.ts (hunk 1 above)
Comment (baltpeter) [reply src/index.ts:470]: Static timeouts across different computers will always be a problem… But that should be fixed in appstraction, shouldn't it?

Comment (zner0L) [reply src/index.ts:470]: I thought so first, too. But actually, I would want it to wait for a device much shorter than for a lenghty emulator startup.

Comment (zner0L) [reply src/index.ts:470]: And since we know we wait for an emulator here, we should extend the timeout, shouldn‘t we?

Comment (baltpeter) [reply src/index.ts:470]: I still feel like there should be a more idiomatic way of waiting for an emulator in appstraction, but sure.

**Pull request merged**